src/js/tasks-function.js

```diff
@@ -0,0 +1,181 @@
+      label[elem.index].setAttribute('id', elem.index);
+      label[elem.index].contentEditable = true;
+      const inputBox = [];
+      inputBox[elem.index] = document.createElement('input');
+      inputBox[elem.index].setAttribute('type', 'checkbox');
+      inputBox[elem.index].setAttribute('id', elem.index);
+      inputBox[elem.index].classList.add('checkbox');
+      const moveBtn = [];
+      moveBtn[elem.index] = document.createElement('button');
+      moveBtn[elem.index].setAttribute('id', elem.index);
+      moveBtn[elem.index].innerHTML = '<i class="fa-solid fa-ellipsis-vertical ellipsis"></i>';
+
+      li[elem.index].append(
+        inputBox[elem.index],
+        label[elem.index],
+        moveBtn[elem.index],
+      );
+      taskList.append(li[elem.index]);
+      li[elem.index].addEventListener('click', () => {
+        li[elem.index].style.backgroundColor = '#ff98008f';
+      });
+      label[elem.index].addEventListener('click', (e) => {
+        e.target.nextSibling.innerHTML = '<i class="fa-solid fa-trash-can"></i>';
+        e.target.nextSibling.style.cursor = 'pointer';
+
+        e.target.nextSibling.addEventListener('click', () => {
+          li[elem.index].remove();
+          this.removetask(elem.index);
+        });
+      });
+      li[elem.index].addEventListener('mouseleave', (e) => {
+        this.updateTask(e.target.id, e.target.innerText);
+      });
+      inputBox[elem.index].addEventListener('change', (e) => {
+        const status = new TaskStatus();
+        if (e.target.checked === true) {
+          status.checked(this.taskData[elem.index]);
+        } else {
+          status.unchecked(this.taskData[elem.index]);
+        }
+        this.updateTask(
+          e.target.nextSibling.id,
+          e.target.nextSibling.innerText,
+        );
+      });
+      if (this.taskData[elem.index].completed === true) {
+        inputBox[elem.index].setAttribute('checked', 'checked');
+        li[elem.index].classList.add('checked');
+        label[elem.index].style.textDecoration = 'line-through';
+      } else if (this.taskData[elem.index].completed === false) {
+        inputBox[elem.index].removeAttribute('checked');
+        label[elem.index].style.textDecoration = 'none';
+      }
+    });
+  };
+
+  addtask = (description, completed, index) => {
+    const newtask = new Tasks(description, completed, index);
+    this.taskData.push(newtask);
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+    }
+  }
+
+  removetask = (item) => {
+    const key = item;
+    if (this.taskData.length === 1) {
+      this.taskData = [];
+    } else {
+      this.taskData.splice(key, 1);
+    }
+    this.taskData.forEach((elem, index) => {
+      elem.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      this.displayTask();
+    }
+  }
```

Comment:
- [ ] After removing a task, it would be nice to set the index of the last item to the length of the array. If an array has one item, the `index` key of the item will be equal to `1`. Similarly, if it has two items, the `index` key of the second item will be equal to `2` while that of the first item will be equal to `1`. In the same vein, an array of ten items will have its tenth item have an index of `10`. To achieve this, kindly do this:

```JavaScript
 this.taskData.forEach((elem, index) => {
      elem.index = index + 1;
    });
```

Indexing in the loop always starts from `0`. 💻 👍